server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -953,6 +953,24 @@ private static MappingsVersion extractVersionFromMappings(String mappings) {
         return new MappingsVersion(value, Objects.hash(properties));
     }
 
+    /**
+     * An accurate node version is no longer required in system index mappings metadata.
+     * because the mappings version should be used to determine if an upgrade is required,
+     * not the node version. However, some parts of the code are still relying on
+     * <code>mappingsNodeVersion</code>. This method allows sections of the code to stop
+     * accurately setting node version in their mappings while other sections continue to
+     * use it. Once all uses of <code>mappingsNodeVersion</code> are removed this method
+     * can be removed too.
+     */
+    @Deprecated
+    private static Version bestEffortExtractNodeVersionFromMappings(String mappings, String versionMetaKey) {
+        try {
+            return extractNodeVersionFromMappings(mappings, versionMetaKey);
```

Comment:
I think it would be better to deprecate this as well so we don't forget to remove it